### ISSUE TITLE
MagicFusion: Boosting Text-to-Image Generation Performance by Fusing Diffusion Models

### ISSUE THREAD

**Issue by nmhkahn** [nmhkahn/paper-reading#74]
## Motivation
- Pre-trained 모델과 fine-tune 모델을 앙상블. CFG를 보니 이게 saliency 기반의 guidance와 비슷해서 이걸 응용

## Method
<img width="974" alt="Screenshot 2023-03-28 at 8 31 31 PM" src="https://user-images.githubusercontent.com/16869368/228222815-3ccf0b74-4075-4f06-8f2e-f7015cafbb88.png">

- <img width="350" alt="Screenshot 2023-03-28 at 8 32 32 PM" src="https://user-images.githubusercontent.com/16869368/228223055-1b8cc692-14bb-4927-895d-bb9b18643ab2.png">
- CFG에서 중요한 영역은 high response인 것을 관찰하였음. 그래서 이걸 활용해서 salience map을 생성
- <img width="35-" alt="Screenshot 2023-03-28 at 8 33 15 PM" src="https://user-images.githubusercontent.com/16869368/228223226-92046c1d-399f-48e5-a638-4571b0b8f78f.png">
    - Blur를 사용해서 high-frequency 노이즈를 제거하고 adjacent region에 대해 response leverage
    - #67 방법에서 attention map을 비슷한 motivation을 기반으로 Blur 사용
- General 모델, expert 모델에서 생성한 salience map을 각각 $\Lambda^g, \Lambda^e$라고 했을 때,
    - <img width="200" alt="Screenshot 2023-03-28 at 8 36 53 PM" src="https://user-images.githubusercontent.com/16869368/228224084-7ab6bdf2-f3ed-4682-b51e-a8b33a5fa205.png">
    - Temperature를 사용한 softmax로 normalize
    - <img width="200" alt="Screenshot 2023-03-28 at 8 37 18 PM" src="https://user-images.githubusercontent.com/16869368/228224178-1f7014e2-475a-4744-bbe3-8b902f654055.png">
    - <img width="200" alt="Screenshot 2023-03-28 at 8 37 25 PM" src="https://user-images.githubusercontent.com/16869368/228224205-03f1acf6-274f-4751-befe-6b55ed6fbb67.png">

**Comment by nmhkahn** [comment]
## Discussion
- Inference 할때 timestep이 진행되면서 알아서 맞춰질 수는 있지만 saliency-aware noise blending이 너무 나이브하다는 인상을 지울 수가 없음
    - G의 prompt가 "A car in the desert" 이고, E가 "white sedan..." 일 때, G와 E가 생성하는 자동차의 위치가 다를거고, 또 argmax를 취한다고 하더라도 E의 자동차 영역이 항상 이긴다는 보장이 없는데 어떻게 저게 된다고 확신할 수 있지
- $\epsilon$ 말고 h-space에서 blending?
    - UNet 디코더 안태워도 됨
